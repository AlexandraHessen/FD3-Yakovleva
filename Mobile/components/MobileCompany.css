 .ProductsGrid{

    border-collapse: collapse;
}

.ProductsGrid .Header{
    border-bottom: solid #e4e4e4 2px;
    border-top: solid #e4e4e4 1px;
    padding: 10px
} 
