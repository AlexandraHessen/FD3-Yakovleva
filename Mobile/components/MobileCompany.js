import React from "react";
import PropTypes from 'prop-types';

import "./MobileCompany.css";

import MobileClient from "./MobileClient";
import ClientEditAdd from "./ClientEditAdd";
import ClientCard from "./ClientCard";

class MobileCompany extends React.PureComponent{
    static propTypes = {
        companyName: PropTypes.string.isRequired,
        clients:PropTypes.array.isRequired,
      };

    state ={
        companyName: this.props.companyName,
        clients: this.props.clients,
        // companyName: this.props.companyName.slice(),
        selectedProductCode: null,
        cardMode: 0,  // 0 - ничего не выводим, 1 - просмотр карточки товара, 2 - редактирование и создание товара
        isEdit: false, // = true когда карточка  в режиме редактирования
        isDelete: false,
        isValid: true, 
        isChanged: false, //проверяем были ли внесены какие-либо изменения в форму, чтобы блокировать действия; когда = true, buttons disabled= true
        isAdd: false, //проверяем когда добавляем товар чтобы рендерить в компоненте соответствующие кнопки и текст для Add
    }

    setName1 = () => {
        this.setState({companyName:'Velcom'});
      };
    
      setName2 = () => {
        this.setState({companyName:'МТС'});
      };

    cbChanged=(changed)=>{
        this.setState({isChanged: changed})
    }

    cbShowCard =(code)=>{
        if (!this.state.isChanged){
            this.setState({
                cardMode: 1, 
                selectedProductCode: code, 
                isEdit: false
            })
        }
    }

    cbEdit=(code)=>{
        if (!this.state.isChanged){
            this.setState({
                cardMode: 2, 
                selectedProductCode: code,
                isEdit: true
            })
        }
    }

    add=()=>{
        if (!this.state.isChanged && !this.state.isEdit){
            let code=this.state.clients.length
            this.setState({
                selectedProductCode: ++code,
                cardMode: 2, 
                isAdd: true,
            })
        }
    }

    cbSave=(editRow)=>{
        let editGoods=this.state.goods.map(row=>(row.code==editRow.code)?editRow:row)
        this.setState({
            cardMode: 0, 
            goods: editGoods,
            isEdit: false
        })
    }

    cbAdd=(newRow)=>{
        this.state.goods.push(newRow)
        this.setState({
            cardMode: 0, 
        })
    }

    cbDelete=(code)=>{
        if (!this.state.isChanged && !this.state.isEdit){
            if(confirm('Вы действительно хотите удалить товар?')){
                let result=this.state.goods;
                result=result.filter(s=>(s.code!==code));
                this.setState({
                    cardMode: 0,
                    goods: result,
                    isDelete: true
                })
            } 
        }
    }

    cbCancel=()=>{
        this.setState({ cardMode: 0,
                        isEdit: false})
    }

    render(){
// ----------------------- ТАБЛИЦА КЛИЕНТОВ -----------------------//   
        var clientsCode=this.state.clients.map( client=>{
            let FIO={surname:client.surname, name:client.name, patronymic:client.patronymic};
            return  <MobileClient key={client.code} 
                        row={client} 
                        code={client.code} 
                        selectedProductCode={this.state.selectedProductCode}
                        cbShowCard={this.cbShowCard}
                        cbEdit={this.cbEdit}
                        cbDelete={this.cbDelete}
                        isChanged={this.state.isChanged}
                        isEdit={this.state.isEdit}
            />
        }

        );

// СТРОКА ДЛЯ РАБОТЫ
        let selectedClientRow=this.state.clients.find((v, i)=>v.code==this.state.selectedProductCode)

        return (
        <div>
            <input type="button" value="Velcom" onClick={this.setName1} />
            <input type="button" value="МТС" onClick={this.setName2} />
            <div >Компания: {this.state.companyName}</div>
            <table className='ProductsGrid'>
                <tbody>
                    <tr> 
                        <th className='Header'>Фамилия</th>
                        <th className='Header'>Имя</th>
                        <th className='Header'>Отчество</th>
                        <th className='Header'>Баланс</th>
                        <th className='Header'>Статус</th>
                        <th className='Header'>Редактировать</th>
                        <th className='Header'>Удалить</th>
                    </tr>
                    {clientsCode}
                </tbody>
            </table>
            <input type='button' value='Добавить клиента' onClick={this.add} disabled={this.state.isChanged||this.state.isEdit}/>

{/*----------------------- ПРОСМОТР КАРТОЧКИ КЛИЕНТА -----------------------*/}            
            {
                (this.state.cardMode=="1") &&
                <ClientCard row={selectedClientRow} 
                />
            }

{/*----------------------- РЕДАКТИРОВАНИЕ И СОЗДАНИЕ -----------------------*/}
            {
                (this.state.cardMode=="2") &&
                <ClientEditAdd key={this.state.selectedProductCode} 
                            code={this.state.selectedProductCode} 
                            row={selectedClientRow} 
                            cbSave={this.cbSave}
                            cbAdd={this.cbAdd}
                            cbChanged={this.cbChanged}
                            cbCancel={this.cbCancel}
                />
            }
        </div>
        )
    }
};

export default MobileCompany;