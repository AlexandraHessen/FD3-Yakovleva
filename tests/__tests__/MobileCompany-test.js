"use strict"
import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from "../components/MobileCompany"
                            //импортируем файл который будем тестировать

let clientsArr = [{code: 1, surname: "Иванов", name: "Иван", patronymic: "Иванович", balance: 200},
                  {code: 2, surname: "Сидоров", name: "Сидор", patronymic: "Сидорович", balance: 250},
                  {code: 3, surname: "Петров", name: "Петр", patronymic: "Петрович", balance: 180},
                  {code: 4, surname: "Григорьев", name: "Григорий", patronymic: "Григорьевич", balance: -220}]
                  // -------- нужен для создания компонента --------//

test("работа кнопок Filter", ()=>{
                            // создаём тестовую версию компонента который будем тестировать
  const component = renderer.create(
    <MobileCompany  clients={clientsArr}/> // с нужными данными для построения
  );

                            // 1. получаем снэпшот (HTML-снимок) компонента для сверки, чтобы вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

                            // найдём в вёрстке компонента саму кнопку
                            // в верстке <input type="button" value="Все" onClick={this.allClients}/>
    const buttonAll = component.root.find( el => el.type=='input' && el.props.value == 'Все' ); 
                            // нажимаем на кнопку
    buttonAll.props.onClick();

                            // 2. получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

                            // "нажмём" кнопку ещё раз
    buttonAll.props.onClick();
  
                            // 3. и получаем окончательный снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

                            // найдём в вёрстке компонента саму кнопку
                            // в верстке <input type="button" value="Активные" onClick={this.activeClients}/> 
                            const buttonActive = component.root.find( el => el.type=='input' && el.props.value == 'Активные' ); 
                            // нажимаем на кнопку
    buttonActive.props.onClick();

                            // 2. получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

                            // "нажмём" кнопку ещё раз
    buttonActive.props.onClick();
  
                            // 3. и получаем окончательный снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

                            // найдём в вёрстке компонента саму кнопку
                            // в верстке <input type="button" value="Заблокированные" onClick={this.blockedClients}/> 
                            const buttonBlock = component.root.find( el => el.type=='input' && el.props.value == 'Заблокированные' ); 
                            // нажимаем на кнопку
    buttonBlock.props.onClick();

                            // 2. получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

                            // "нажмём" кнопку ещё раз
    buttonBlock.props.onClick();
  
                            // 3. и получаем окончательный снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
}

)