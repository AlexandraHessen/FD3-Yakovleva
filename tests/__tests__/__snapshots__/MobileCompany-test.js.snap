// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа кнопок Filter 1`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="ProductsGrid"
  >
    <tbody>
      <tr>
        <th
          className="Header"
        >
          Фамилия
        </th>
        <th
          className="Header"
        >
          Имя
        </th>
        <th
          className="Header"
        >
          Отчество
        </th>
        <th
          className="Header"
        >
          Баланс
        </th>
        <th
          className="Header"
        >
          Статус
        </th>
        <th
          className="Header"
        >
          Редактировать
        </th>
        <th
          className="Header"
        >
          Удалить
        </th>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Иванов
        </td>
        <td
          className="InfoGoods"
        >
          Иван
        </td>
        <td
          className="InfoGoods"
        >
          Иванович
        </td>
        <td
          className="InfoGoods"
        >
          200
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Сидоров
        </td>
        <td
          className="InfoGoods"
        >
          Сидор
        </td>
        <td
          className="InfoGoods"
        >
          Сидорович
        </td>
        <td
          className="InfoGoods"
        >
          250
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Петров
        </td>
        <td
          className="InfoGoods"
        >
          Петр
        </td>
        <td
          className="InfoGoods"
        >
          Петрович
        </td>
        <td
          className="InfoGoods"
        >
          180
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Григорьев
        </td>
        <td
          className="InfoGoods"
        >
          Григорий
        </td>
        <td
          className="InfoGoods"
        >
          Григорьевич
        </td>
        <td
          className="InfoGoods"
        >
          -220
        </td>
        <td
          className="blocked InfoGoods"
        >
          blocked
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="buttonAdd"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа кнопок Filter 2`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="ProductsGrid"
  >
    <tbody>
      <tr>
        <th
          className="Header"
        >
          Фамилия
        </th>
        <th
          className="Header"
        >
          Имя
        </th>
        <th
          className="Header"
        >
          Отчество
        </th>
        <th
          className="Header"
        >
          Баланс
        </th>
        <th
          className="Header"
        >
          Статус
        </th>
        <th
          className="Header"
        >
          Редактировать
        </th>
        <th
          className="Header"
        >
          Удалить
        </th>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Иванов
        </td>
        <td
          className="InfoGoods"
        >
          Иван
        </td>
        <td
          className="InfoGoods"
        >
          Иванович
        </td>
        <td
          className="InfoGoods"
        >
          200
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Сидоров
        </td>
        <td
          className="InfoGoods"
        >
          Сидор
        </td>
        <td
          className="InfoGoods"
        >
          Сидорович
        </td>
        <td
          className="InfoGoods"
        >
          250
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Петров
        </td>
        <td
          className="InfoGoods"
        >
          Петр
        </td>
        <td
          className="InfoGoods"
        >
          Петрович
        </td>
        <td
          className="InfoGoods"
        >
          180
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Григорьев
        </td>
        <td
          className="InfoGoods"
        >
          Григорий
        </td>
        <td
          className="InfoGoods"
        >
          Григорьевич
        </td>
        <td
          className="InfoGoods"
        >
          -220
        </td>
        <td
          className="blocked InfoGoods"
        >
          blocked
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="buttonAdd"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа кнопок Filter 3`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="ProductsGrid"
  >
    <tbody>
      <tr>
        <th
          className="Header"
        >
          Фамилия
        </th>
        <th
          className="Header"
        >
          Имя
        </th>
        <th
          className="Header"
        >
          Отчество
        </th>
        <th
          className="Header"
        >
          Баланс
        </th>
        <th
          className="Header"
        >
          Статус
        </th>
        <th
          className="Header"
        >
          Редактировать
        </th>
        <th
          className="Header"
        >
          Удалить
        </th>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Иванов
        </td>
        <td
          className="InfoGoods"
        >
          Иван
        </td>
        <td
          className="InfoGoods"
        >
          Иванович
        </td>
        <td
          className="InfoGoods"
        >
          200
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Сидоров
        </td>
        <td
          className="InfoGoods"
        >
          Сидор
        </td>
        <td
          className="InfoGoods"
        >
          Сидорович
        </td>
        <td
          className="InfoGoods"
        >
          250
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Петров
        </td>
        <td
          className="InfoGoods"
        >
          Петр
        </td>
        <td
          className="InfoGoods"
        >
          Петрович
        </td>
        <td
          className="InfoGoods"
        >
          180
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Григорьев
        </td>
        <td
          className="InfoGoods"
        >
          Григорий
        </td>
        <td
          className="InfoGoods"
        >
          Григорьевич
        </td>
        <td
          className="InfoGoods"
        >
          -220
        </td>
        <td
          className="blocked InfoGoods"
        >
          blocked
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="buttonAdd"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа кнопок Filter 4`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="ProductsGrid"
  >
    <tbody>
      <tr>
        <th
          className="Header"
        >
          Фамилия
        </th>
        <th
          className="Header"
        >
          Имя
        </th>
        <th
          className="Header"
        >
          Отчество
        </th>
        <th
          className="Header"
        >
          Баланс
        </th>
        <th
          className="Header"
        >
          Статус
        </th>
        <th
          className="Header"
        >
          Редактировать
        </th>
        <th
          className="Header"
        >
          Удалить
        </th>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Иванов
        </td>
        <td
          className="InfoGoods"
        >
          Иван
        </td>
        <td
          className="InfoGoods"
        >
          Иванович
        </td>
        <td
          className="InfoGoods"
        >
          200
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Сидоров
        </td>
        <td
          className="InfoGoods"
        >
          Сидор
        </td>
        <td
          className="InfoGoods"
        >
          Сидорович
        </td>
        <td
          className="InfoGoods"
        >
          250
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Петров
        </td>
        <td
          className="InfoGoods"
        >
          Петр
        </td>
        <td
          className="InfoGoods"
        >
          Петрович
        </td>
        <td
          className="InfoGoods"
        >
          180
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="buttonAdd"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа кнопок Filter 5`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="ProductsGrid"
  >
    <tbody>
      <tr>
        <th
          className="Header"
        >
          Фамилия
        </th>
        <th
          className="Header"
        >
          Имя
        </th>
        <th
          className="Header"
        >
          Отчество
        </th>
        <th
          className="Header"
        >
          Баланс
        </th>
        <th
          className="Header"
        >
          Статус
        </th>
        <th
          className="Header"
        >
          Редактировать
        </th>
        <th
          className="Header"
        >
          Удалить
        </th>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Иванов
        </td>
        <td
          className="InfoGoods"
        >
          Иван
        </td>
        <td
          className="InfoGoods"
        >
          Иванович
        </td>
        <td
          className="InfoGoods"
        >
          200
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Сидоров
        </td>
        <td
          className="InfoGoods"
        >
          Сидор
        </td>
        <td
          className="InfoGoods"
        >
          Сидорович
        </td>
        <td
          className="InfoGoods"
        >
          250
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Петров
        </td>
        <td
          className="InfoGoods"
        >
          Петр
        </td>
        <td
          className="InfoGoods"
        >
          Петрович
        </td>
        <td
          className="InfoGoods"
        >
          180
        </td>
        <td
          className="active InfoGoods"
        >
          active
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="buttonAdd"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа кнопок Filter 6`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="ProductsGrid"
  >
    <tbody>
      <tr>
        <th
          className="Header"
        >
          Фамилия
        </th>
        <th
          className="Header"
        >
          Имя
        </th>
        <th
          className="Header"
        >
          Отчество
        </th>
        <th
          className="Header"
        >
          Баланс
        </th>
        <th
          className="Header"
        >
          Статус
        </th>
        <th
          className="Header"
        >
          Редактировать
        </th>
        <th
          className="Header"
        >
          Удалить
        </th>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Григорьев
        </td>
        <td
          className="InfoGoods"
        >
          Григорий
        </td>
        <td
          className="InfoGoods"
        >
          Григорьевич
        </td>
        <td
          className="InfoGoods"
        >
          -220
        </td>
        <td
          className="blocked InfoGoods"
        >
          blocked
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="buttonAdd"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа кнопок Filter 7`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="ProductsGrid"
  >
    <tbody>
      <tr>
        <th
          className="Header"
        >
          Фамилия
        </th>
        <th
          className="Header"
        >
          Имя
        </th>
        <th
          className="Header"
        >
          Отчество
        </th>
        <th
          className="Header"
        >
          Баланс
        </th>
        <th
          className="Header"
        >
          Статус
        </th>
        <th
          className="Header"
        >
          Редактировать
        </th>
        <th
          className="Header"
        >
          Удалить
        </th>
      </tr>
      <tr
        className="ProductRow"
      >
        <td
          className="InfoGoods"
        >
          Григорьев
        </td>
        <td
          className="InfoGoods"
        >
          Григорий
        </td>
        <td
          className="InfoGoods"
        >
          Григорьевич
        </td>
        <td
          className="InfoGoods"
        >
          -220
        </td>
        <td
          className="blocked InfoGoods"
        >
          blocked
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="EditButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td
          className="InfoGoods"
        >
          <input
            className="DelButton"
            disabled={undefined}
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="buttonAdd"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;
